internal/api/v1/application/part.go

```diff
@@ -118,8 +118,38 @@ func fetchAppChart(c *gin.Context, ctx context.Context, logger logr.Logger, clus
 		return apierror.InternalError(err)
 	}
 
+	logger.Info("input", "chart archive", chartArchive)
+
+	// Try to read the archive as local path first, before falling back to retrieval via http.
+
+	file, err := os.Open(chartArchive)
+	if err == nil {
+		logger.Info("input", "chart archive", "is file")
+
+		fileInfo, err := file.Stat()
+		if err == nil {
+			logger.Info("input", "chart archive", "has stat")
+
+			contentLength := fileInfo.Size()
+			contentType := "application/x-gzip"
+
+			logger.Info("input", "chart archive", "returning file")
+
+			c.DataFromReader(http.StatusOK, contentLength, contentType, bufio.NewReader(file),
+				map[string]string{
+					"X-Content-Length": strconv.FormatInt(contentLength, 10),
+				})
+
+			return nil
+		}
+	}
+
+	logger.Info("input", "chart archive", "retrieving by http")
+
 	response, err := http.Get(chartArchive) // nolint:gosec // app chart repo ref
 	if err != nil || response.StatusCode != http.StatusOK {
+		logger.Info("fail, http issue")
+
 		c.Status(http.StatusServiceUnavailable)
 		return nil
```

Comment:
I'm not sure, why we are not returning an error? Is that maybe something that was wrong?

Reply:
Not sure actually. And no, that was not what was wrong.
What was wrong was trying to `http.Get()` a file, i.e. `/assets/epinio-application-0.1.24.tgz`.